File: server/service/murmur.go
package service

import (
	"server/global"
	"server/model/database"
	"server/model/request"
)

type MurmurService struct{}

func (m *MurmurService) MurmurCreate(mur database.Murmur) error {
	err := global.DB.Create(&mur).Error
	if err != nil {
		return err
	}
	return nil
}

func (m *MurmurService) MurmurDelete(ids request.MurmurDelete) error {
	err := global.DB.Where("id IN ?", ids.Ids).Delete(&database.Murmur{}).Error
	if err != nil {
		return err
	}
	return nil
}

func (m *MurmurService) MurmurList() ([]database.Murmur, error) {
	var murmurList []database.Murmur
	err := global.DB.Order("created_at desc").Find(&murmurList).Error
	if err != nil {
		return []database.Murmur{}, err
	}
	return murmurList, nil
}

File: web/src/api/murmur.ts
import type {ApiResponse} from "@/utils/request";
import service from "@/utils/request";

export interface Murmur {
    id: number;
    content: string;
    created_at: string;
    updated_at: string;
}

export const murmurAll = (): Promise<ApiResponse<Murmur[]>> => {
    return service({
        url: '/murmur/all',
        method: 'get',
    })
}

export interface MurmurCreateRequest {
    content: string;
}

export const murmurCreate = (data: MurmurCreateRequest): Promise<ApiResponse<undefined>> => {
    return service({
        url: '/murmur/create',
        method: 'post',
        data: data,
    })
}

export interface MurmurDeleteRequest {
    ids: number[];
}

export const murmurDelete = (data: MurmurDeleteRequest): Promise<ApiResponse<undefined>> => {
    return service({
        url: '/murmur/delete',
        method: 'delete',
        data: data,
    })
}
